src/algorithm/boundingbox.rs

```diff
@@ -0,0 +1,209 @@
+use num::{Float};
+
+use types::{Bbox, Point, MultiPoint, LineString, MultiLineString, Polygon, MultiPolygon};
+
+/// Calculation of the bounding box of a geometry.
+
+pub trait BoundingBox<T: Float> {
+    /// Return a Bounding Box of a geometry
+    ///
+    /// ```
+    /// use geo::{Point, LineString, Coordinate};
+    /// use geo::algorithm::boundingbox::BoundingBox;
+    ///
+    /// let mut vec = Vec::new();
+    /// vec.push(Point::new(40.02f64, 116.34));
+    /// vec.push(Point::new(42.02f64, 116.34));
+    /// vec.push(Point::new(42.02f64, 118.34));
+    /// let linestring = LineString(vec);
+    /// let bbox = linestring.bbox().unwrap();
+    ///
+    /// println!("Bbox top left coordinates: {}, {}", bbox.xmin, bbox.ymax);
+    /// println!("Bbox bottom right coordinates: {}, {}", bbox.xmax, bbox.ymin);
+    /// ```
+    ///
+    fn bbox(&self) -> Option<Bbox<T>>;
+}
+
+fn get_bbox<T>(vect: &Vec<Point<T>>) -> Option<Bbox<T>>
+    where T: Float
```

Comment:
On Mon, Jul 18, 2016 at 9:16 AM, Guillaume P. <notifications@github.com>
wrote:

> In src/algorithm/boundingbox.rs
> <https://github.com/georust/rust-geo/pull/41#discussion_r71106626>:
>
> > +fn get_bbox<T>(vect: &Vec<Point<T>>) -> Option<Bbox<T>>
> > +    where T: Float
>
> You can rewrite this function to take an IntoIterator<Item = &Point<T>>.
> Then, multisomething is just a flat_map.
>

Sorry I don't understand how I should modify the function, any hints?